src/storage/storage.cc

```diff
@@ -333,9 +333,11 @@ Status Storage::OpenForReadOnly() { return Open(true); }
 
 Status Storage::CreateBackup() {
   LOG(INFO) << "[storage] Start to create new backup";
-  std::lock_guard<std::mutex> lg(backup_mu_);
+  std::lock_guard<std::mutex> lg(config_->backup_mu_);
+  // Note: `config_->backup_dir` may change, so we should use same dir during the task.
+  std::string task_backup_dir = config_->backup_dir;
```

Comment:
Hi. I have a question, it seems that `config_->backup_dir` is always protected by `config_->backup_mu_`, is it necessary to store it in a temporary variable here?

Reply:
Yes, at the begining I want to use a shorter critical section, but currently it leave it here.